tools/notes-2019-06-25.md

```diff
@@ -0,0 +1,58 @@
+        * SP: Receiver?
+        * DR: foo.bar?.(1,2, 3) you could do the same with .call but you have to rebind the this. 
+        * SP: It's not clear maybe that you would have to rebind the this, that might be confusing. The pitfall I saw from Ruby is that it is nice to be able to write this, but as codebases change, it seems to bitrot earlier because it becomes a silent failure.  It is somewhat like Rusts .unwrap?
+        * DR: I'm biased--I'd say, throw a type system at it. The feature isn't error suppression, but rather fixing missing values. If you want to protect against bitrot, you need a tool to help. 
+        * SP: When I was writing TS the other day, I was wishing I had this. Anyway, I think ?.( is a good idea, since I don't want people to have to use ?.call( and remember this. Ruby gets around thisby the default being the other way around.
+    * Pipeline: Significant research ongoing
+        * DE: <summarizes the research efforts led by yulia>.  We all in TC39 are really excited about this for language design
+        * SP: is there a place to read about the research
+        * DE: there was a blog post introducing it, I don't know if there has been a writeup on the intermediate status, but the end plan is to create/write an academic paper.
+    * Standard library/built-in modules: Standing controversy about split vs unified namespaces
+        * DE: <summarizes discussion/controversy about layering and who can set/govern>
+        * SP: I would like to talk about standard library functions that require a receiver.
+        * DE: it keeps being suggested that there were a way to do this with the standard library.
+        * SP: Like Promise.resolve - if we exposed resolve from a module or is it bound from the global Promise constructor
+        * DE: This gets into explaining existing things that way too. There is a promise from Domenic Dinicola about this called 'getOriginals' where the answer to that could be 'yes' but it is not very ergonomic and really about being useful to tooling.  It's pretty ugly, but it makes sense as output for tooling. 
+        * SP: So right now is it fair to say anything about that proposal?
+        * DE: Not really, except that it is being talked about by some Chrome people - it is very early.
+        * GB: The one thing that is really nice about this proposal is that you can guarantee that these haven't been mutated.  This should lead to better optimizations in build tools
+        * DE: there is a shallow guarantee, but not a deep one. Specifically the imported identifiers themselves cannot be modified, but members on them can be mutated.  `import resolve from ‘std:reflect’; resolve will be the same, but resolve.foo = true can change.
```

Comment:
this is missing a backtick; it messes up syntax highlighting after it.

Reply:
thanks